include/pybind11/cast.h

```diff
@@ -308,6 +309,7 @@ template <typename type> class type_caster_base : public type_caster_generic {
 };
 
 template <typename type, typename SFINAE = void> class type_caster : public type_caster_base<type> { };
+template <typename type> using make_caster = type_caster<intrinsic_t<type>>;
```

Comment:
For consistency, it would be nice to this ``make_caster`` alias in all places where ``type_caster<typename intrinsic_type<type>::type>`` is currently used (quite a few I think)